src/operator/mxnet_op.h

```diff
@@ -345,6 +394,13 @@ __global__ void mxnet_generic_kernel(int N, Args... args) {
   }
 }
 
+template<typename OP, typename ...Args>
+__global__ void mxnet_generic_kernel_ex(int N, Args... args) {
+  for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < N; i += blockDim.x * gridDim.x) {
```

Comment:
Can this calculation "blockDim.x * gridDim.x" be moved outside of the loop so it doesn't execute so often (assuming the compiler doesn't somehow know blockDim.x and gridDim.x won't change, and optimize it out)?